deploy/samples/monitoring-stack.yaml

```diff
@@ -7,4 +7,4 @@ spec:
   retention: 1d
   resourceSelector:
     matchLabels:
-      system: foo
+      app.kubernetes.io/part-of: monitoring-stack-operator
```

Comment:
I updated this to `app.kubernetes.io/part-of: monitoring-stack-operator` since resourceSelector label need to match the object label while creating prometheus cr's, I either needed to add `system:foo` to cr's or just update the label here. Looking for better way to handle if this is not the right way

Reply:
Hi JP, I wonder if we the stacks should collect these service monitors. I would expect them to end up in some other prometheus since they're related to the operator itself. So I don't think we should worry too much about matching the selectors of the stack resource and the service monitor(s) for the operator.

Reply:
sure I added that since I was trying to test alert I will remove that. We want to keep only rules file?

Reply:
I think it's fine to keep the ServiceMonitor as well. Let's just make sure it selects the component label from the pods, not just the `app.kubernetes.io/name`:
https://github.com/rhobs/monitoring-stack-operator/blob/main/deploy/operator/monitoring-stack-operator-deployment.yaml#L18-L20

Reply:
updated, I generated bundle locally it works fine

I looked at https://docs.openshift.com/container-platform/4.9/operators/understanding/olm-packaging-format.html#olm-bundle-format_olm-packaging-format servicemonitor and prometheusrule are supported for bundle

So a question is since the service, servicemonitor and prometheusrule will be only used when prometheus resource is created deploying them by default is ok?

Reply:
I think deploying them by default is okay, we just have to make sure that they don't get picked up by some prometheus deployed by a monitoring stack. 

We should probably document how to collect these monitoring resources in a Prometheus instance, I think SRE will definitely be interested in that.

Reply:
Without a matching resourceSelector label it doesn't get picked up by default, cant we use that logic? I meant while creating monitoringstack cr we can specify the label we should use as in this sample-monitoring stack file?

Reply:
Since the goal of this PR is to generate alerts if **MSO** fails to reconcile, should we worry about the stacks we deploy? Not only that, these rules don't apply to stack we deploy. Hence ... we shouldn't need to modify the sample stack - cr

Reply:
The prometheus resources servicemonitor, prometheusrule were not picking by operator unless the resourceSelector matching label is there. I had updated to test this change but I can remove.

Reply:
I reverted it to same as before but to confirm we can use this logic to make sure it doesn't picked by monitoring-stack-operator by default unless a matchingLabel found?

Reply:
@fpetkovski wdyt?